### PR TITLE
MT-755: Migrate react-native-musora-forum package to Function Components

### PR DESCRIPTION
**Pull request by MeenuDhanoa**

### CODE REVIEW

File: src/commons/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+        tintColor={appColor}
+        progressBackgroundColor={appColor}
+        onRefresh={refresh}
+        refreshing={refreshing}
+      />
+    ),
+    [appColor, refresh, refreshing]
+  );
+
+  const keyExtractor = useCallback((item: ISearchItem) => item.id.toString(), []);
```
Comment (csillaj) [inline src/commons/Search.tsx]: No need for useCallback

File: src/components/Threads.tsx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+  const flEmpty = useMemo(
+    () => (
+      <Text style={styles.emptyList}>
+        {tab ? 'You are not following any threads.' : 'No threads.'}
+      </Text>
+    ),
+    [styles.emptyList, tab]
+  );
+
+  const keyExtractor = useCallback((item: number) => item.toString(), []);
```
Comment (csillaj) [inline src/components/Threads.tsx]: no need for useCallback

File: src/commons/Post.tsx [hunk 3]
```diff
@@ -0,0 +1,347 @@
+import React, { FunctionComponent, useCallback, useState } from 'react';
```
Comment (csillaj) [inline src/commons/Post.tsx]: When I open a post the app crashes with this error "Text strings must be rendered within a Text component

Comment (csillaj) [reply src/commons/Post.tsx]: I think it's because of line 303 on HtmlRenderer
You need to add !!htmlWidth

File: src/commons/Post.tsx [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+          {
+            ...post,
+            content: `<blockquote><b>${post?.author?.display_name}</b>:<br>${post?.content}</blockquote>`,
+          },
+        ],
+      });
+    }
+  }, [navigate, onPostCreated, post]);
+
+  return (
```
Comment (csillaj) [inline src/commons/Post.tsx:132]: Something is off with how post are displayed when a user replies to a post or multiquotes multiple quotes, see images (1st is from your branch, 2nd from master)
image
image

File: src/components/CRUD.tsx [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+      pop();
+    } else if (postId) {
+      await deletePost(postId).request;
+      onDeleteProp?.(postId);
+    }
+    goBack();
+  }, [goBack, pop, postId, threadId, type, onDeleteProp]);
+
+  return (
+    <SafeAreaView
```
Comment (csillaj) [inline src/components/CRUD.tsx:250]: Something is off here too, take a look how this page should look when replying to a post
image
When I test this with your changes  the original post to which I'm replying is not showing

**Review by csillaj — CHANGES_REQUESTED** [review]

File: src/commons/Post.tsx (hunk 4 above)
Comment (MeenuDhanoa) [reply src/commons/Post.tsx:132]: Fixed it.

File: src/components/CRUD.tsx (hunk 5 above)
Comment (MeenuDhanoa) [reply src/components/CRUD.tsx:250]: Fixed it

**Review by csillaj — APPROVED** [review]

**Review by DidierF — APPROVED** [review]

**Pull request merged**